src/Generator/EnumGenerator/EnumGenerator.php

```diff
@@ -0,0 +1,146 @@
+        $this->methods = $methods;
+    }
+
+    public function generate(): string
+    {
+        $output = '';
+
+        if (null !== $this->name->getNamespace()) {
+            $output .= 'namespace ' . $this->name->getNamespace() . ';' . self::LINE_FEED . self::LINE_FEED;
+        }
+
+        $output .= 'enum ' . $this->name->getName() . $this->retrieveType() . ' {'
+            . self::LINE_FEED
+            . $this->retrieveCases()
+            . $this->retrieveMethods()
+            . '}'
+            . self::LINE_FEED;
+
+        return $output;
+    }
+
+    private function retrieveType(): string
+    {
+        if ($this->cases->getBackedType() === null) {
+            return '';
+        }
+
+        return ': ' . ($this->cases->getBackedType() ?? '');
+    }
+
+    private function retrieveCases(): string
+    {
+        return implode(
+            '',
+            array_map(
+                fn (string $case): string => self::INDENTATION . 'case ' . $case . ';' . self::LINE_FEED,
+                $this->cases->getCases()
+            )
+        );
+    }
+
+    /**
+     * @psalm-param array{
+     *      name: string,
+     *      pureCases?: list<string>,
+     *      backedCases?: array{
+     *          type: string,
+     *          cases: array<string, int|string>
+     *      },
+     *      methods?: list<array{
+     *          name: string,
+     *          visibility?: string,
+     *          static?: bool,
+     *          returntype?: string,
+     *      }>
+     * } $options
+     */
+    public static function withConfig(array $options): self
+    {
+        return new self(
+            Name::fromName($options['name']),
+            CaseFactory::fromOptions($options),
+            self::fromMethods($options),
+        );
+    }
+
+    /**
+     * @psalm-param array{
+     *      name: string,
+     *      pureCases?: list<string>,
+     *      backedCases?: array{
+     *          type: string,
+     *          cases: array<string, int|string>
+     *      },
+     *      methods?: list<array{
+     *          name: string,
+     *          visibility?: string,
+     *          static?: bool,
+     *          returntype?: string,
+     *      }>
+     * } $options
+     * @return list<MethodGenerator>
+     */
+    private static function fromMethods(array $options): array
+    {
+        if (array_key_exists('methods', $options) === false) {
+            return [];
+        }
+
+        return array_map(
+            static fn (array $methodOptions): MethodGenerator => MethodGenerator::fromArray($methodOptions),
+            $options['methods'],
+        );
+    }
+
+    private function retrieveMethods(): string
+    {
+        if ($this->methods === []) {
+            return '';
+        }
+
+        $methodsOutput = '';
+        foreach ($this->methods as $method) {
+            $methodsOutput .= self::LINE_FEED . $method->generate();
+        }
+
+        return $methodsOutput;
+    }
+}
```

Comment:
One method missing here is `::fromReflection()`, operating on a given enum reflection instance - that's one of the biggest utilities in this generator system.

Reply:
I'll work on adding this method, but how can we do that while remaining compatible with 7.4?

From what I understand in the RFC, the reflection we need to use for that is `ReflectionEnum` but that is only included in 8.1.

Reply:
I'll take a look at the suggestion

Reply:
I implemented the `::fromReflection()` method.

I updated the Psalm baseline for now, since I have no knowledge how to tackle the specific issues that arose from it, especially the `ParseError` ones, which makes sense since the analysis is done in 8.0.